[NAOT][Apple][mobile] Undefined symbols for architecture x86_64: do_vxsort_avx2 causing AppleAppBuilderTask to fail

## Build Information
Build: https://dev.azure.com/dnceng-public/public/_build/results?buildId=629964
Build error leg or test failing: 
- tvossimulator-x64 Release AllSubsets_NativeAOT_RuntimeTests
- iossimulator-x64 Release AllSubsets_NativeAOT_RuntimeTests
<!-- Error message template  -->
## Error Message

Fill the error message using [step by step known issues guidance](https://github.com/dotnet/arcade/blob/main/Documentation/Projects/Build%20Analysis/KnownIssueJsonStepByStep.md).

<!-- Use ErrorMessage for String.Contains matches. Use ErrorPattern for regex matches (single line/no backtracking). Set BuildRetry to `true` to retry builds with this error. Set ExcludeConsoleLog to `true` to skip helix logs analysis. -->

```json
{
  "ErrorMessage": "/Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: Undefined symbols for architecture x86_64: [/Users/runner/work/1/s/src/tests/build.proj]",
  "BuildRetry": false,
  "ExcludeConsoleLog": false
}
```


Error log:
```
normal/x86_64/nativeaot_SmokeTests_UnitTests_dependency_info.dat -o /Users/runner/work/1/s/artifacts/tests/coreclr/obj/iossimulator.x64.Release/Managed/build/iOSApps/nativeaot_SmokeTests_UnitTests/bin-nativeaot_SmokeTests_UnitTests-x64/nativeaot_SmokeTests_UnitTests/Debug-iphonesimulator/nativeaot_SmokeTests_UnitTests.app/nativeaot_SmokeTests_UnitTests [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3624320Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: Undefined symbols for architecture x86_64: [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3624960Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:   "do_vxsort_avx2(unsigned char**, unsigned char**, unsigned char*, unsigned char*)", referenced from: [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3625650Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:       WKS::gc_heap::plan_phase(int) in libRuntime.WorkstationGC.a(gcwks.cpp.o) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3626330Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:   "do_vxsort_avx512(unsigned char**, unsigned char**, unsigned char*, unsigned char*)", referenced from: [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3627020Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:       WKS::gc_heap::plan_phase(int) in libRuntime.WorkstationGC.a(gcwks.cpp.o) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3627650Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:   "IsSupportedInstructionSet(InstructionSet)", referenced from: [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3628360Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:       WKS::gc_heap::grow_mark_list() in libRuntime.WorkstationGC.a(gcwks.cpp.o) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3629010Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:       WKS::gc_heap::plan_phase(int) in libRuntime.WorkstationGC.a(gcwks.cpp.o) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3629730Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:       WKS::gc_heap::do_post_gc() in libRuntime.WorkstationGC.a(gcwks.cpp.o) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3630350Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:   "InitSupportedInstructionSet(int)", referenced from: [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3631020Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:       WKS::gc_heap::initialize_gc(unsigned long, unsigned long, unsigned long) in libRuntime.WorkstationGC.a(gcwks.cpp.o) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3631690Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: ld: symbol(s) not found for architecture x86_64 [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3632460Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: clang: error: linker command failed with exit code 1 (use -v to see invocation) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3633540Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: note: Run script build phase 'Generate CMakeFiles/ALL_BUILD' will be run during every build because the option to run the script phase "Based on dependency analysis" is unchecked. (in target 'ALL_BUILD' from project 'nativeaot_SmokeTests_UnitTests') [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3634320Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: ** BUILD FAILED ** [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3634840Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: The following build commands failed: [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3636140Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: 	Ld /Users/runner/work/1/s/artifacts/tests/coreclr/obj/iossimulator.x64.Release/Managed/build/iOSApps/nativeaot_SmokeTests_UnitTests/bin-nativeaot_SmokeTests_UnitTests-x64/nativeaot_SmokeTests_UnitTests/Debug-iphonesimulator/nativeaot_SmokeTests_UnitTests.app/nativeaot_SmokeTests_UnitTests normal (in target 'nativeaot_SmokeTests_UnitTests' from project 'nativeaot_SmokeTests_UnitTests') [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3637100Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: (1 failure) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3637870Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:    at Utils.RunProcess(TaskLoggingHelper logger, String path, String args, IDictionary`2 envVars, String workingDir, Boolean ignoreErrors, Boolean silent, MessageImportance debugMessageImportance) in /_/src/tasks/Common/Utils.cs:line 123 [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3638790Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:    at Xcode.BuildAppBundle(String xcodePrjPath, Boolean optimized, String devTeamProvisioning) in /_/src/tasks/AppleAppBuilder/Xcode.cs:line 644 [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3639540Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:    at AppleAppBuilderTask.Execute() in /_/src/tasks/AppleAppBuilder/AppleAppBuilder.cs:line 334 [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3640320Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3641250Z /Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/Users/runner/work/1/s/src/tests/build.proj]
2024-04-03T21:32:33.3641830Z     0 Warning(s)
2024-04-03T21:32:33.3641990Z     6 Error(s)
```
<!--Known issue error report start -->

### Report
|Build|Definition|Step Name|Console log|Pull Request|
|---|---|---|---|---|
|[632885](https://dev.azure.com/dnceng-public/public/_build/results?buildId=632885)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/632885/logs/339)|dotnet/runtime#100389|
|[631365](https://dev.azure.com/dnceng-public/public/_build/results?buildId=631365)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/631365/logs/732)|dotnet/runtime#100389|
|[630062](https://dev.azure.com/dnceng-public/public/_build/results?buildId=630062)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/630062/logs/644)|dotnet/runtime#100389|
|[629964](https://dev.azure.com/dnceng-public/public/_build/results?buildId=629964)|dotnet/runtime|Build Tests|[Log](https://dev.azure.com/dnceng-public/cbb18261-c48f-4abb-8651-8cdcb5474649/_apis/build/builds/629964/logs/375)||
#### Summary
|24-Hour Hit Count|7-Day Hit Count|1-Month Count|
|---|---|---|
|0|4|4|
<!--Known issue error report end -->
<!-- Known issue validation start -->
 ### Known issue validation
**Build: :mag_right:** https://dev.azure.com/dnceng-public/public/_build/results?buildId=629964
**Error message validated:** `[/Users/runner/work/1/s/src/mono/msbuild/apple/build/AppleBuild.targets(296,5): error MSB4018: Undefined symbols for architecture x86_64: [/Users/runner/work/1/s/src/tests/build.proj]`]
**Result validation:** :white_check_mark: Known issue matched with the provided build.
**Validation performed at:** 4/4/2024 7:53:18 PM UTC
<!-- Known issue validation end -->

--- a/src/tests/Directory.Build.props
+++ b/src/tests/Directory.Build.props
@@ -212,6 +212,7 @@
 
   <PropertyGroup Condition="'$(RuntimeFlavor)' == 'coreclr' and '$(TargetsAppleMobile)' == 'true'">
     <_targetOS>$(TargetOS)</_targetOS>
+    <_targetArchitecture>$(TargetArchitecture)</_targetArchitecture>
     <UseNativeAOTRuntime>true</UseNativeAOTRuntime>
     <NativeLib>static</NativeLib>
     <CustomNativeMain>true</CustomNativeMain>

